### PR TITLE
use shared gh action

### PR DESCRIPTION
**Pull request by olamy**

### CODE REVIEW

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -16,37 +16,18 @@
...
-        run: mvn install javadoc:javadoc -e -B -V 
+  build:
+    name: Build it
+    uses: codehaus-plexus/.github/.github/workflows/maven.yml@master
+    with:
+      jdk-fast-fail-build: '11'
+      jdk-matrix: '["11", "17"]'
+      jdk-distribution-matrix: '["zulu", "temurin", "microsoft", "liberica"]'
+      os-matrix: '["ubuntu-latest","windows-latest", "macOS-latest"]'
+      maven_args: 'install javadoc:javadoc -e -B -V -fae '
```
Comment (slawekjaranowski) [inline .github/workflows/maven.yml:33]: why need install?

File: .github/workflows/maven.yml [hunk 2]
```diff
@@ -16,37 +16,18 @@
...
-      - name: Build with Maven
-        run: mvn install javadoc:javadoc -e -B -V 
+  build:
+    name: Build it
+    uses: codehaus-plexus/.github/.github/workflows/maven.yml@master
+    with:
+      jdk-fast-fail-build: '11'
+      jdk-matrix: '["11", "17"]'
+      jdk-distribution-matrix: '["zulu", "temurin", "microsoft", "liberica"]'
+      os-matrix: '["ubuntu-latest","windows-latest", "macOS-latest"]'
```
Comment (slawekjaranowski) [inline .github/workflows/maven.yml:32]: the same as default

File: .github/workflows/maven.yml (hunk 1 above)
Comment (olamy) [reply .github/workflows/maven.yml:33]: why not? it's a temporary node so doesn't hurt anybody  :)

File: .github/workflows/maven.yml (hunk 2 above)
Comment (olamy) [reply .github/workflows/maven.yml:32]: default has more jdks

Comment (olamy) [reply .github/workflows/maven.yml:32]: @slawekjaranowski  And default in master are changed without PRs neither asking around

**Pull request merged**